core/src/main/java/org/fenixedu/messaging/domain/Message.java

```diff
@@ -254,14 +314,75 @@ public void delete() {
 
     public Set<String> getBccs() {
         Set<String> base = recipientsToEmails(getBccSet());
-        if (getExtraBccs() != null && !getExtraBccs().isEmpty()) {
-            Set<String> extended = Sets.newHashSet(getExtraBccs().replace(',', ' ').replace(';', ' ').split("\\s+"));
-            extended.addAll(base);
-            return extended;
-        }
+        base.addAll(getExtraBccsSet());
         return base;
     }
 
+    public Set<String> getExtraBccsSet() {
+        String extraBccs = getExtraBccs();
+        if (!Strings.isNullOrEmpty(extraBccs)) {
+            return Sets.newHashSet(getExtraBccs().replace(',', ' ').replace(';', ' ').split("\\s+"));
+        } else {
+            return Sets.newHashSet();
+        }
+    }
+
+    private static Map<Locale, Set<String>> emailsByLocale(Set<PersistentGroup> groups, Predicate<String> emailValidator) {
+        Map<Locale, Set<String>> emails = new HashMap<Locale, Set<String>>();
+        Locale defLocale = I18N.getLocale();
```

Comment:
I think this should be `Locale.getDefault()` instead. My thinking assumes that if a user has no preferred locale, he/she is satisfied with the default, and as such, that should be the one in which the message is sent, not the one in which the sender is currently in.